Provide self contained npm scripts

KaneFreeman opened the pull request:
**Type:** feature

The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code has been formatted with [`prettier`](https://prettier.io/) as per the [readme code style guidelines](./../#code-style)
* [ ] Unit or Functional tests are included in the PR

<!--
Our bots should ensure:

* [ ] All contributors have signed a CLA
* [ ] The PR passes CI testing
* [ ] Code coverage is maintained
* [ ] The PR has been reviewed and approved
-->

**Description:**
Adds in dev, build and test scripts, along with adding @dojo/cli as a dev dependency, to the package.json templates.

Resolves #154

aciccarello commented:
I would recommend not using the abbreviated flags to make the scripts more intuitive for new users.

KaneFreeman commented:
@aciccarello the scripts have been updated with their long formats.

KaneFreeman commented:
@aciccarello I have updated the scripts to their long form.

agubler commented:
@KaneFreeman I guess we should add something about the scripts to the template README files?

KaneFreeman commented:
I have updated the README files to include mention of the new scripts, while still mentioning the `@dojo/cli` equivalent. I added a `build:dev` script in as well since it is explicitly called out in the READMEs.

agubler commented on src/templates/ts/skeleton/README.md:
```diff
@@ -4,29 +4,37 @@ This project was generated with the [Dojo CLI](https://github.com/dojo/cli) & [D
-To run the unit tests against built bundles, first the run a test build with `dojo build --mode unit`. The build test artifacts are written to the `output/tests/unit` directory.
+To run the unit tests against built bundles, run `npm run test:unit`.
+
+To be re-run the unit tests without needing to re-build the full application each time, first build the app with `--most unit` and the `--watch` option, `dojo build --mode unit --watch`. Then run `dojo test --config local` to run the unit tests as needed.
```
Should this be `--mode unit`?

KaneFreeman replied:
Lol, yeah. I have updated that.

agubler commented on src/run.ts:
```diff
@@ -17,6 +17,15 @@ export default async function(helper: Helper, args: CreateAppArgs) {
+		dev: 'dojo build --mode dev --watch file --serve',
+		build: 'dojo build --mode dist',
+		buildDev: `dojo build --mode dev`,
+		test: 'npm run test:unit && npm run test:functional',
```
I think that `npm run test` should translate to `dojo test` that doesn't need a bundle to be built as it does transpilation on the fly.

Perhaps the others:
 * `npm run test:unit` should be `dojo build --mode unit && dojo test --unit --config local` 
 * `npm run test:functional` should be `dojo build --mode functional && dojo test --functional --config local` 
 * `npm run test:all` should be `dojo build --mode unit && dojo build --mode functional && dojo test --all --config local`

KaneFreeman replied:
@agubler The proposed changes have been made.
